### PR TITLE
feat: add resource iteration events

### PR DESCRIPTION
**Pull request by oliverbl**
* see https://docs.gitlab.com/ee/api/resource_iteration_events.html

### CODE REVIEW

File: tests/unit/objects/test_resource_iteration_events.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            content_type="application/json",
+            status=200,
+        )
+        yield rsps
+
+
+def test_list_project_issue_iteration_events(
+    project_issue, resp_list_project_issue_iteration_events
+):
+    iteration_events = project_issue.resourceiterationevents.list()
```
Comment (nejch) [inline tests/unit/objects/test_resource_iteration_events.py:42]: To go with the above:
```suggestion
    iteration_events = project_issue.resource_iteration_events.list()
```

File: tests/unit/objects/test_resource_iteration_events.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    iteration_event = iteration_events[0]
+    assert isinstance(iteration_event, ProjectIssueResourceIterationEvent)
+    assert iteration_event.resource_type == "Issue"
+
+
+def test_get_project_issue_iteration_event(
+    project_issue, resp_get_project_issue_iteration_event
+):
+    iteration_event = project_issue.resourceiterationevents.get(1)
```
Comment (nejch) [inline tests/unit/objects/test_resource_iteration_events.py:53]:
```suggestion
    iteration_event = project_issue.resource_iteration_events.get(1)
```

**Review by nejch — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution @oliverbl, very clean PR! Just a tiny note on style. Also would you be able to add a sentence in the examples here?

https://github.com/python-gitlab/python-gitlab/blob/main/docs/gl_objects/projects.rst#examples

**Comment by oliverbl** [comment]
> Thanks a lot for the contribution @oliverbl, very clean PR! Just a tiny note on style. Also would you be able to add a sentence in the examples here?
> 
> https://github.com/python-gitlab/python-gitlab/blob/main/docs/gl_objects/projects.rst#examples


I'm not sure which is the correct place for the documentation of iteration events. Currenty iteration events can only be accessed by project issues, so I added the example to the project issue documentation.

But also other events (like Milestone events) have their own documentation file

**Comment by nejch** [comment]
> > Thanks a lot for the contribution @oliverbl, very clean PR! Just a tiny note on style. Also would you be able to add a sentence in the examples here?
> > https://github.com/python-gitlab/python-gitlab/blob/main/docs/gl_objects/projects.rst#examples
> 
> I'm not sure which is the correct place for the documentation of iteration events. Currenty iteration events can only be accessed by project issues, so I added the example to the project issue documentation.
> 
> But also other events (like Milestone events) have their own documentation file

Ah sorry, I sent the wrong link. That makes sense in the issues docs @oliverbl, if the events endpoints grow we can later create a separate page for them.

**Review by nejch — APPROVED** [review]

**Pull request merged**